packages/docs-infra/src/pipeline/loadPrecomputedCodeHighlighter/parseFunctionParameters.test.ts

```diff
@@ -0,0 +1,155 @@
+import { describe, it, expect } from 'vitest';
+import { parseFunctionParameters } from './parseFunctionParameters';
+
+describe('parseFunctionParameters', () => {
```

Comment:
I haven't noticed a test case for parsing functino params with default val, e.g.:
```
function test(a = {}, b = "")
```

Reply:
The function being parsed in the factory is:
```
const func = createFunc(a, b)
```
so this wouldn't be valid code:
```
const func = createFunc(a = {}, b = {})
```

With your feedback, I took a closer look and improved the handling of the parsing to support this which wasn't working:
```
const func = createFunc(a as boolean, b as string)
```

Now I'm realizing we are actually parsing function arguments, not parameters, so I've renamed the function to be more clear.